python/ray/data/tests/test_distinct.py

```diff
@@ -0,0 +1,118 @@
+import pyarrow as pa
+import pytest
+
+import ray
+
+
+@pytest.fixture
+def sample_data():
+    """Basic dataset with duplicates for testing."""
+    return pa.table({"a": [1, 2, 2, 1, 3], "b": ["x", "y", "y", "x", "z"]})
+
+
+@pytest.fixture
+def subset_data():
+    """Dataset for testing distinct on subset of columns."""
+    return pa.table({"a": [1, 1, 2, 3, 2, 4], "b": ["x", "y", "y", "z", "w", "foo"]})
+
+
+@pytest.fixture
+def empty_data():
+    """Empty dataset with schema for testing edge cases."""
+    return pa.table({"a": [], "b": []})
+
+
+def test_distinct_across_blocks(sample_data):
```

Comment:
for all tests, can you please make sure you use one of those ray data fixtures to avoid cluster leakage? like shutdown_only